src/views/studio/CreateEditChannelView/CreateEditChannelView.tsx

```diff
@@ -319,21 +316,37 @@ export const CreateEditChannelView: React.FC<CreateEditChannelViewProps> = ({ ne
       }
     }
 
-    const completed = await handleTransaction({
-      preProcess: processAssets,
-      txFactory: (updateStatus) =>
-        newChannel
-          ? joystream.createChannel(activeMemberId, metadata, assets, updateStatus)
-          : joystream.updateChannel(activeChannelId ?? '', activeMemberId, metadata, assets, updateStatus),
-      onTxFinalize: uploadAssets,
-      onTxSync: refetchDataAndCacheAssets,
-      successMessage: {
-        title: newChannel ? 'Channel successfully created!' : 'Channel successfully updated!',
-        description: newChannel
-          ? 'Your channel was created and saved on the blockchain. Feel free to start using it!'
-          : 'Changes to your channel were saved on the blockchain.',
-      },
-    })
+    const setContentIds = async ({ avatarId, coverId }: ContentIdCbArgs) => {
+      avatarContentId = avatarId
+      coverContentId = coverId
+    }
+
+    const instance = await joystream
+
+    const completed =
+      instance &&
+      (await handleTransaction({
+        preProcess: processAssets,
+        txFactory: async (updateStatus) =>
+          newChannel
+            ? await instance.createChannel(activeMemberId, metadata, assets, proxy(updateStatus), proxy(setContentIds))
+            : await instance.updateChannel(
+                activeChannelId ?? '',
+                activeMemberId,
+                metadata,
+                assets,
+                proxy(updateStatus),
```

Comment:
I'm wondering about those `proxy` calls, I'm not sure if they should be inside our components, seems like internal workings are leaking. Although I guess they are necessary when callbacks are involved. I'm considering potentially having wrappers for all the joystream functions in one place (provider?) and we only use `comlink` there so it doesn't affect the components. WDYT?